scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1235,7 +1249,7 @@ def run(self):
                 else:
                     inst.metrics.update(self.instances[inst.name].metrics)
                     inst.metrics["handler_time"] = inst.execution_time
-                    inst.metrics["unrecognized"] = []
+                    inst.metrics.setdefault("unrecognized", [])
```

Comment:
There is a question whether this is something we want. Does every result report need `"unrecognized": []`? Would leaving that key out be worse? Of course, any change could break scripts that assume that the `"unrecognized"` key is always present.